src/NonStrippingCreditCard.php

```diff
@@ -28,4 +35,100 @@ public function setNumber($value)
          */
         return $this->setParameter('number', $value);
     }
+
+    /**
+     * Sets the payment instrument name.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setPaymentInstrumentName($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('paymentInstrumentName', $value);
+    }
+
+    /**
+     * Gets the payment instrument name.
+     *
+     * @return string|null
+     */
+    public function getPaymentInstrumentName()
+    {
+        return $this->getParameter('paymentInstrumentName');
+    }
+
+    /**
+     * Sets the payment network.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setPaymentNetwork($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('paymentNetwork', $value);
+    }
+
+    /**
+     * Gets the payment network.
+     *
+     * @return string|null
+     */
+    public function getPaymentNetwork()
+    {
+        return $this->getParameter('paymentNetwork');
+    }
+
+    /**
+     * Sets the transaction identifier.
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setTransactionIdentifier($value)
```

Comment:
@parth21shah I thought we talked about not returning this since it didn't seem particularly useful and it's awkward to have a transaction-level ID on a payment method object? we've already got transaction IDs from vindicia

Reply:
This is a token that ApplePay sends and I thought we need this to make sure when we are updating buildPayamentMethod it can handle all the parameters that we want to send or vindicia sends back

Reply:
we only need to support parameters that are useful!

Reply:
i still don't think we need this parameter. is there any reason that we do?

Reply:
We have to send this in auth call otherwise it won't work. We decided not to save it on our side which is okay but have to tell this to vindicia

Reply:
@parth21shah the example you provided in the tech spec indicates that the transaction identifier is part of the payment data token already. do we really have to pass it to vindicia twice? do you have any docs about that?

Reply:
It doesn't look like the same transaction id.

Reply:
Changing the method names to setApplePayTransactionReference and getApplePayTransactionReference per discussion. May be able to eliminate it completely if need be, after further testing.